client/src/Hooks/useCommentsFetch.js

```diff
@@ -1,14 +1,24 @@
 import { useState, useEffect } from 'react';
 import api from '../api.js';
-export default function useCommentsFetch (parent, returnWithComments = true) {
+export default function useCommentsFetch (parent, setPopup, returnWithComments = true) {
 	const [comments, setComments] = useState([]);
 
 	useEffect(() => {
-		api.sendPostCommentsRequest(parent, returnWithComments).then((res) => {
-			setComments(res.data);
-		}).catch((error) =>
-			console.log(error)
-		);
+		async function fetchData () {
+			try {
+				const res = await api.sendPostCommentsRequest(parent, returnWithComments);
+				if (res.status === 200) {
+					setComments(res.data);
+				} else {
+					setPopup({ message: 'Something went wrong when fetching the comments' });
+				}
+			} catch (e) {
+				setPopup({ message: 'Something went wrong when fetching the comments' });
+				console.log(e);
+			}
+		}
+		fetchData();
+	// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What dependencies have been omitted and why?

Reply:
the setPopup function as this function will not change

Reply:
Since this is the props, so you should add it in the dependency anyway. And if you're sure, that it won't be changed, it won't influence the behaviour. But if it did change, it would save you from bigger bugs.
And I would check inside the useEffect if comments exist already. It depends on what you're going to do if setPopup is undefined. 

By the way, this issue won't be so important at all if you have Typescript. Because right now you might really reuse this function with the changing setPopup (initially undefined and them populated). And no one will catch this. You must write the code imaging that some other person gonna make changes without deeply reading the logic.

Right it is not so critical. But I really ask you to verify all your code in the future using this rule: QA will gonna break your code eventually. Don't let this happen, this is your reputation and this is how you will grow as a specialist.
And your future colleagues will reuse your code and make mistakes and spend hours to debug. 

This is the example of bug I fixed yesterday - someone forgot to add error (one of the output that you should save, not only data from API) to the dependencies and fetching never stopped if there was an error. And I relied on this code supposing it is safe to use, but it is hard to debug the edge cases.